### PR TITLE
Document Improvement Plan Task3&4: Bundle Tutorial and QuickStart Page

### PR DESCRIPTION
**Pull request by tangy5**
Hi,

This is the third draft PR for improving the QuickStart page, and with the new tutorial for bundle app use in MONAI Label. 

I'm thinking of the two tutorials for the quickstart page, one for Radiology, and one for Bundle. 
Users can choose one to start. 
There are step-by-step instructions on loading the bundle app with MONAI Label server, start 3D Slicer, add plugin, and use the bundle model for inference and training. Also with a customized bundle usage. 

The edited quickstart page: 

https://github.com/tangy5/MONAILabel/blob/docImprove_t3t4/docs/source/quickstart.rst

Let me know if you have any comments. Welcome to discuss here!

@SachidanandAlle @Nic-Ma

### CODE REVIEW

File: docs/source/quickstart.rst [hunk 1]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+4. Config describes flexible workflow and components, allows for different low-level python implementations
+
+Currently available bundles: `Model-Zoo <https://github.com/Project-MONAI/model-zoo>`_
+
+.. _MONAI Label with 3DSlicer:
+
+Prerequisite Setup
+=================================
+
+1. Install MONAI Label and 3DSlicer
```
Comment (jamesobutler) [inline docs/source/quickstart.rst:340]: `3DSlicer` -> `3D Slicer` in all places. There is a space in the name.

**Comment by diazandr3s** [comment]
This is great @tangy5 
[This part ](https://github.com/tangy5/MONAILabel/blob/docImprove_t3t4/docs/source/quickstart.rst#annotating-a-custom-dataset) has an error on the command to execute.

> monailabel apps --name deepedit --download --output .

It should be `monailabel apps --name radiology --download --output .`

File: docs/source/quickstart.rst [hunk 2]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+
+1. Self-contained model package with all the necessary information
+
+2. Structured config that easy to override or reconstruct the workflow
+
+3. Config provides good readability and usability by separating parameter settings from the python code
+
+4. Config describes flexible workflow and components, allows for different low-level python implementations
+
+Currently available bundles: `Model-Zoo <https://github.com/Project-MONAI/model-zoo>`_
```
Comment (Nic-Ma) [inline docs/source/quickstart.rst:333]: Please use this link: https://github.com/Project-MONAI/model-zoo/releases/tag/hosting_storage_v1

Thanks.

File: docs/source/quickstart.rst [hunk 3]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+  * Advancing heterogeneous dataset (e.g., CT, MRI, Pathology, etc) with corresponding pre- and post-processing modules.
+  * Ready-to-Use inference of hundreds of anatomies (e.g., multi-organ abdominal segmentation, whole-brain segmentation) with trained model checkpoints.
+  * Deploying robust interactive labeling tools such as DeepEdit.
+
+.. _Model Zoo for MONAI Label:
+
+Model Zoo for MONAI Label
+-----------------------------
+
+MONAI Model Zoo hosts a collection of medical imaging models in the MONAI Bundle format.
```
Comment (Nic-Ma) [inline docs/source/quickstart.rst:319]: Please add the link of Model Zoo:
MONAI [Model Zoo](https://github.com/Project-MONAI/model-zoo) hosts a collection of medical imaging models in the MONAI Bundle format.

Thanks.

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+Introduction:
+===============
+
+Customized Networks for MONAI Label
+------------------------------------
+
+This tutorial introduces the usage of the Bundle app in MONAILabel  - **monaibundle**.
+
+The Bundle App empowers MONAILabel with customized models, pre- and post-processing, and any anatomies for labeling tasks.
+The Bundle App supports various training/inference frameworks with `Model-Zoo <https://github.com/Project-MONAI/model-zoo>`_
```
Comment (Nic-Ma) [inline docs/source/quickstart.rst:305]: I don't quite understand this line, maybe better:
The Bundle App supports various [bundle](https://docs.monai.io/en/latest/mb_specification.html) based training / inference pipelines within the [Model-Zoo](https://github.com/Project-MONAI/model-zoo).

Thanks.

File: docs/source/quickstart.rst [hunk 5]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+.. _Model Zoo for MONAI Label:
+
+Model Zoo for MONAI Label
+-----------------------------
+
+MONAI Model Zoo hosts a collection of medical imaging models in the MONAI Bundle format.
+All source code of models (bundles) are tracked in models/, and for each distinct version of a bundle,
+it will be archived as a .zip file (named in the form of bundle_name_version.zip) and stored in Releases.
+
+The monaibundle defines the model package and supports building python-based workflows via structured configurations
```
Comment (Nic-Ma) [inline docs/source/quickstart.rst:323]: monaibundle -> MONAI bundle

File: docs/source/quickstart.rst [hunk 6]
```diff
@@ -275,3 +284,164 @@ At this point OHIF can be used to annotate the data in the DICOM server via the
...
+  * Ready-to-Use inference of hundreds of anatomies (e.g., multi-organ abdominal segmentation, whole-brain segmentation) with trained model checkpoints.
+  * Deploying robust interactive labeling tools such as DeepEdit.
+
+.. _Model Zoo for MONAI Label:
+
+Model Zoo for MONAI Label
+-----------------------------
+
+MONAI Model Zoo hosts a collection of medical imaging models in the MONAI Bundle format.
+All source code of models (bundles) are tracked in models/, and for each distinct version of a bundle,
```
Comment (Nic-Ma) [inline docs/source/quickstart.rst:320]: We should not expect users to check the source code of `model-zoo` repo, please use the release link of models: https://github.com/Project-MONAI/model-zoo/releases/tag/hosting_storage_v1

Thanks.

**Review by Nic-Ma — COMMENTED** [review]
Hi @tangy5 ,

Thanks for the nice tutorial, besides the inline comments, I have 2 more comments:
1. For the introduction of MONAI bundle, I would suggest to redirect to the web page: https://docs.monai.io/en/latest/mb_specification.html.
Don't put the slide in the doc.
3. May I know why remove the copyright of many files?

Thanks.

**Comment by zmsunnyday** [comment]
Hi @tangy5, very clear tutorials. Great work! Here are some minor suggestions:

- After downloading the spleen dataset, we should remove the files with prefix of "_" && suffix of "nii.gz" in the folders, otherwise it might cause problems when visualizing in 3d Slicer.
- I got problem when using DICOM server as data source. When starting the monailabel server by using `monailabel start_server --app radiology --studies http://locahost:8042/dicom-web --conf models deepedit --username orthanc --password orthanc`, an error was raised: usage: monailabel [-h] {start_server,apps,datasets,plugins} ...
monailabel: error: unrecognized arguments: --username orthanc --password orthanc

- 
-

File: docs/source/quickstart.rst (hunk 1 above)
Comment (tangy5) [reply docs/source/quickstart.rst:340]: Thanks, added a space to all 3D Slicer.

**Comment by tangy5** [comment]
> 

Thanks, fixed in this version.

File: docs/source/quickstart.rst (hunk 2 above)
Comment (tangy5) [reply docs/source/quickstart.rst:333]: Thanks, changed.

File: docs/source/quickstart.rst (hunk 3 above)
Comment (tangy5) [reply docs/source/quickstart.rst:319]: Added the link, thanks.

File: docs/source/quickstart.rst (hunk 2 above)
Comment (tangy5) [reply docs/source/quickstart.rst:333]: Changed, thanks

File: docs/source/quickstart.rst (hunk 5 above)
Comment (tangy5) [reply docs/source/quickstart.rst:323]: Changed, thanks.

File: docs/source/quickstart.rst (hunk 6 above)
Comment (tangy5) [reply docs/source/quickstart.rst:320]: yes, good suggestion. Changed the Model zoo link to the release link of models.

**Comment by tangy5** [comment]
> 

Hey @zmsunnyday, thanks for providing these comments. These user experiences are really critical!
I'm continuing the investigation of the tutorial. 

For 1: Are you suggesting removing "_" to all NIFTI data? Could you point me to an example for not loading .nii.gz image correctly? Simply removing _ might cause a further problem as indices are typically used in the filename. Thanks. 

On 2: There should be no argument or some confusion regarding --username --password, let me test more Dicom data and update this command. 

3: Thanks for the comment. I updated a line for manually downloading bundle files and using the customized scripts. The error might be because of not finding the bundle app's customized script.  Currently, using a customized scripts bundle needs these steps for directing environment to the hybrid components:

```
cd <path to the bundle model>/renalStructures_UNEST_segmentation_v0.1.0
# add customized scripts in the downloaded bundle
export PYTHONPATH=$PYTHONPATH:"'/monaibundle/model/renalStructures_UNEST_segmentation_v0.1.0/scripts"
```
Thank you again. I will continue to update the Dicom server issue you encountered once figured out.

**Comment by zmsunnyday** [comment]
> > 
> 
> Hey @zmsunnyday, thanks for providing these comments. These user experiences are really critical! I'm continuing the investigation of the tutorial.
> 
> For 1: Are you suggesting removing "_" to all NIFTI data? Could you point me to an example for not loading .nii.gz image correctly? Simply removing _ might cause a further problem as indices are typically used in the filename. Thanks.
> 
> On 2: There should be no argument or some confusion regarding --username --password, let me test more Dicom data and update this command.
> 
> 3: Thanks for the comment. I updated a line for manually downloading bundle files and using the customized scripts. The error might be because of not finding the bundle app's customized script. Currently, using a customized scripts bundle needs these steps for directing environment to the hybrid components:
> 
> ```
> cd <path to the bundle model>/renalStructures_UNEST_segmentation_v0.1.0
> # add customized scripts in the downloaded bundle
> export PYTHONPATH=$PYTHONPATH:"'/monaibundle/model/renalStructures_UNEST_segmentation_v0.1.0/scripts"
> ```
> 
> Thank you again. I will continue to update the Dicom server issue you encountered once figured out.

Thanks @tangy5,

For 1: Only when downloading the data from MSD website manually, there will be files starting with "._", which may cause problems. Downloading with monailabel has no such problem. My fault. Sorry for that.
For 2: When running `monailabel start_server --app radiology --studies http://locahost:8042/dicom-web --conf models deepedit --username orthanc --password orthanc`, an error is raised that  "--username" and "--password" are unrecognized arguments:
```
Using PYTHONPATH=/opt:
usage: monailabel [-h] {start_server,apps,datasets,plugins} ...
monailabel: error: unrecognized arguments: --username orthanc --password orthanc
```
MONAI 0.9.1 and MONAILabel 0.4.2 are installed.
I'll try bundle part later.

Thanks so much!
Meng

**Review by SachidanandAlle — APPROVED** [review]

**Pull request merged**